generator/src/test/java/com/linkedin/pegasus/generator/TestDataTemplateGeneratorCmdLineApp.java

```diff
@@ -178,11 +178,17 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
       Assert.assertTrue(generatedSource.contains("class " + pegasusTypeName),
           "Incorrect generated class name.");
 
-      // Match the generated annotation (use regex since this is error-prone)
+      // Validate the generated annotation (use a strange regex-based comparison since temp files can be weird e.g. /var vs. /private/var)
       final Matcher generatedAnnotationMatcher = GENERATED_ANNOTATION_PATTERN.matcher(generatedSource);
       Assert.assertTrue(generatedAnnotationMatcher.find(), "Unable to find a valid @Generated annotation in the generated source file.");
-      final String expectedGeneratedAnnotation = "Generated from " + expectedTypeNamesToSourceFileMap.get(pegasusTypeName);
-      Assert.assertEquals(generatedAnnotationMatcher.group(1), expectedGeneratedAnnotation, "Unexpected @Generated annotation.");
+      final String expectedPath = expectedTypeNamesToSourceFileMap.get(pegasusTypeName);
+      final String actualPath = generatedAnnotationMatcher.group(1);
```

Comment:
can we give comment examples? Although I know you can be pretty good at Regular Expression, so might be everyone else 

But natural language comments are still appreciated.

Reply:
Good point, this testing logic is a little convoluted. I'll update it